### PR TITLE
Add support for db-audit logging (create and list)

### PR DESCRIPTION
**Pull request by aquatiko**
Summary:

This PR add support for 2 operations of DB-audit: associate a db-audit configuration to a cluster and list assoicated DB audit config on a cluster.

Test Plan:
Manually tested the operations, working as expected.
CLI logs ---

```
[rkumar@dev-server-rkumar-31 ybm-cli]$ ./ybm cluster list
A newer version is available. Please upgrade to the latest version v0.1.18
Name             Tier        Version        State         Health    Provider   Regions     Nodes     Node Res.(Vcpu/Mem/DiskGB/IOPS)
polite-chicken   Dedicated   2.18.5.0-b93   CONFIGURING   ❓        AWS        us-east-1   3         4 / 16GB / 200GB / 3000
[rkumar@dev-server-rkumar-31 ybm-cli]$ ./ybm db-audit list --cluster-id=7bb68af6-0875-42e0-8665-dcf634ed9fd1
A newer version is available. Please upgrade to the latest version v0.1.18
No export config found for cluster-id:7bb68af6-0875-42e0-8665-dcf634ed9fd1


[rkumar@dev-server-rkumar-31 ybm-cli]$ ./ybm db-audit-logs-exporter list --cluster-id=7bb68af6-0875-42e0-8665-dcf634ed9fd1
A newer version is available. Please upgrade to the latest version v0.1.18
ID                                     Date Created               Cluster ID                             Telemetry Provider ID                  State         Ysql Config
a9f395f4-4c19-493e-960a-b597fa68da37   2024-02-18T05:22:36.001Z   7bb68af6-0875-42e0-8665-dcf634ed9fd1   7c07c103-e3b2-48b6-ac30-764e9b5275e1   CONFIGURING   {"log_settings":{"log_catalog":true,"log_client":false,"log_level":"LOG","log_parameter":false,"log_relation":false,"log_statement_once":false},"statement_classes":["READ","WRITE"]}


[rkumar@dev-server-rkumar-31 ybm-cli]$ ./ybm db-audit-logs-exporter assign --cluster-id=02ac0228-1f73-4467-a402-801f98407268 --telemetry-provider-id=7c07c103-e3b2-48b6-ac30-764e9b5275e1 --ysql-config=log_catalog=true,log_client=false,log_level=INFO,log_parameter=true --statement_classes=READ,WRITE
A newer version is available. Please upgrade to the latest version v0.1.18
The db audit exporter config 6bbe9a83-8c78-459e-8f41-9ba3a3119758 is being created
ID                                     Date Created               Cluster ID                             Telemetry Provider ID                  State         Ysql Config
6bbe9a83-8c78-459e-8f41-9ba3a3119758   2024-02-18T06:17:05.889Z   02ac0228-1f73-4467-a402-801f98407268   7c07c103-e3b2-48b6-ac30-764e9b5275e1   CONFIGURING   {"log_settings":{"log_catalog":true,"log_client":false,"log_level":"LOG","log_parameter":false,"log_relation":false,"log_statement_once":false},"statement_classes":["READ","WRITE"]}
```

### CODE REVIEW

**Comment by shinji62** [comment]
Two comments
1 Please add test you should find plenty of example
2 wondering if is not better a command with `logs audit-exporter` or similar.

**Comment by aquatiko** [comment]
> 3 comments 1 Please add test you should find plenty of example 2 wondering if is not better a command with `logs audit-exporter` or similar. 3 Ysql-config is gonna be hard to write for a cli .. Json is hell



> 3 comments 1 Please add test you should find plenty of example 2 wondering if is not better a command with `logs audit-exporter` or similar. 3 Ysql-config is gonna be hard to write for a cli .. Json is hell

@shinji62 will look into the test suite. 

My rationale is that the feature is called db-audit and its a commonly know to devlopers, hence kept that keyword there.
Do you mean `logs audit-event list`? suggestions @jagdish-yb  

Any other suggestion how to handle ysql-config? make every key in the json as a flag?

File: internal/formatter/db_audit_logs_exporter.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	"encoding/json"
+
+	"github.com/sirupsen/logrus"
+	ybmclient "github.com/yugabyte/yugabytedb-managed-go-client-internal"
+)
+
+const (
+	defaultDbAuditLogsExporterListing = "table {{.ID}}\t{{.CreatedAt}}\t{{.ClusterId}}\t{{.TelemetryProviderId}}\t{{.State}}\t{{.YsqlConfig}}"
+	exportConfigCreatedAtHeader       = "Date Created"
+	clusterIdHeader                   = "Cluster ID"
```
Comment (shinji62) [inline internal/formatter/db_audit_logs_exporter.go:28]: this header should exist somewhere else.

File: internal/formatter/db_audit_logs_exporter.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	telemetryProviderIdHeader         = "Telemetry Provider ID"
+	exportConfigState                 = "State"
```
Comment (shinji62) [inline internal/formatter/db_audit_logs_exporter.go:30]: this one as well

File: internal/formatter/db_audit_logs_exporter.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+import (
+	"encoding/json"
+
+	"github.com/sirupsen/logrus"
+	ybmclient "github.com/yugabyte/yugabytedb-managed-go-client-internal"
+)
+
+const (
+	defaultDbAuditLogsExporterListing = "table {{.ID}}\t{{.CreatedAt}}\t{{.ClusterId}}\t{{.TelemetryProviderId}}\t{{.State}}\t{{.YsqlConfig}}"
+	exportConfigCreatedAtHeader       = "Date Created"
```
Comment (shinji62) [inline internal/formatter/db_audit_logs_exporter.go:27]: this one as well

**Review by jagdish-yb — APPROVED** [review]
lgtm

**Comment by shinji62** [comment]
Always confuse by the telemetry-id vs metrics-exporter-id .. Are they the samee ?

cc @cdavid @daniel-yb

**Comment by aquatiko** [comment]
> Always confuse by the telemetry-id vs metrics-exporter-id .. Are they the samee ?
> 
> cc @cdavid @daniel-yb

@shinji62 yes they are the same but we have migrated to `telemetry-id` in other places in ybm and this will soon follow in cli

**Review by shinji62 — APPROVED** [review]

**Pull request merged**